libraries/ms-common/inc/pwm_input.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+// PWM Input Module
+
+#include <stdint.h>
+
+#include "pwm_mcu.h"
+#include "status.h"
+
+typedef void (*PwmInputHandler)(uint32_t dc, uint32_t period, void *context);
```

Comment:
At high frequencies will this constantly run or is it only every few rising edges? If it runs every rising edge you may want to document that it will slow down processing significantly. If it runs fairly often you probably want to poll using a soft timer rather than using a callback as you want interrupts to be very fast and it is possible the callback will do some heavy math.

Reply:
Good point, I changed it so that getters are the only way to get the PWM readings